presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergDistributedQueries.java

```diff
@@ -0,0 +1,244 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.testing.assertions.Assert;
+import com.facebook.presto.tests.AbstractTestDistributedQueries;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.SystemSessionProperties.ITERATIVE_OPTIMIZER_TIMEOUT;
+import static com.facebook.presto.SystemSessionProperties.OPTIMIZER_USE_HISTOGRAMS;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.testing.MaterializedResult.resultBuilder;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.IntStream.range;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public abstract class TestIcebergDistributedQueries
+        extends AbstractTestDistributedQueries
+{
+    private final CatalogType catalogType;
+    private final Map<String, String> extraConnectorProperties;
+
+    protected TestIcebergDistributedQueries(CatalogType catalogType, Map<String, String> extraConnectorProperties)
+    {
+        this.catalogType = requireNonNull(catalogType, "catalogType is null");
+        this.extraConnectorProperties = requireNonNull(extraConnectorProperties, "extraConnectorProperties is null");
+    }
+
+    protected TestIcebergDistributedQueries(CatalogType catalogType)
+    {
+        this(catalogType, ImmutableMap.of());
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        return IcebergQueryRunner.createIcebergQueryRunner(ImmutableMap.of(), catalogType, extraConnectorProperties);
+    }
+
+    @Override
+    protected boolean supportsNotNullColumns()
+    {
+        return false;
+    }
+
+    @Override
+    public void testRenameTable()
+    {
+    }
+
+    @Override
+    public void testRenameColumn()
+    {
+    }
+
+    @Override
+    public void testUpdate()
+    {
+        // Updates are not supported by the connector
+    }
+
+    @Override
+    public void testDescribeOutput()
+    {
+    }
+
+    @Override
+    public void testDescribeOutputNamedAndUnnamed()
+    {
+    }
+
+    @Override
+    public void testDelete()
+    {
+        // Test delete all rows
```

Comment:
can we further split these into individual test methods or does each depend on the previous?

Reply:
After check the code, I find the method `testDelete()` is a special case. When we used this method to override the same method in parent class, Iceberg didn't support MOR row level data deletion yet, so it cannot successfully run the corresponding test method of the parent class. However, now we have supported row level deletion, so we can directly run the `testDelete()` method of the parent class. So I can move this method to `IcebergDistributedTestBase` directly, so that it can be used to test v1 COW based deletion.

The other methods either do not support certain functionalities in the corresponding methods in parent class (such as not support `char(10)`), or modify the timeout compare to the corresponding methods in parent class. It feels that their purpose is not to extend testing methods based on parent class, but is used to demonstrate the comparison with the behavior defined by the parent class test cases. For example, through the override test method, we can intuitively see that Iceberg does not support char (10), but its behavior towards strings is the same as its parent class. Once we move them elsewhere, we actually lose this intuitive comparison. What do you think?

Reply:
Besides, after we improve the timeout issue, we may eventually remove the override of methods `testLargeIn()` and `testLargeInWithHistograms()`.